alacritty_terminal/src/event.rs

```diff
@@ -52,6 +52,9 @@ pub enum Event {
 
     /// Shutdown request.
     Exit,
+
+    /// Child process exited, successfully or not.
+    ChildExit(bool),
```

Comment:
I do not understand what it should be then: 
* https://doc.rust-lang.org/std/process/struct.Child.html#method.try_wait returns `Result<Option<ExitStatus>>`
* https://github.com/alacritty/alacritty/blob/992011a4cd9a35f197acc0a0bd430d89a0d01013/alacritty_terminal/src/tty/windows/mod.rs#L117 has similar ideas: we might disconnect from waiting before registering the fact that the process was stopped, ergo we cannot really go and call `GetExitCodeProcess` for this case?

I also do not want to use https://doc.rust-lang.org/std/io/type.Result.html since I have to potentially map every 421-like error code to some `Error` variant or go with https://doc.rust-lang.org/std/io/struct.Error.html#method.other which is also not good — technically, child process cancellations are not _errors_, just a different exit code returned.


So, I will get back to the `Option<i32>` model on this case and will be glad to see your alternative proposals.